### PR TITLE
Blog post: Deploy to AWS with Terraform

### PR DESCRIPTION
**Pull request by kcoax**
Explaining how to use one-click deployer

### CODE REVIEW

File: blog/2021-07-13-tdex-terraform-automation.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+author_title: Maintainer of TDEX Network
+tags: [terraform, automation, tdex]
+---
+
+Configure and run TDex box easily using Terraform automation. 
+
+<!--truncate-->
+
+
+As a first step it is required toobtain aws access and secret keys. Best practice is to create new user in AWS IAM. Proceed with opening AWS IAM users page, and click on Add user. Provide user with name, and below for the Access type select Programmatic access. Next, make sure that your AWS account(access key) has all required privileges to create EC2 instances and S3 access.
```
Comment (tiero) [inline blog/2021-07-13-tdex-terraform-automation.md:14]:
```suggestion
As a first step it is required to obtain aws access and secret keys. Best practice is to create new user in AWS IAM. Proceed with opening AWS IAM users page, and click on Add user. Provide user with name, and below for the Access type select Programmatic access. Next, make sure that your AWS account(access key) has all required privileges to create EC2 instances and S3 access. 
```

File: blog/2021-07-13-tdex-terraform-automation.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
+---
+slug: tdex-terraform-automation
+title: Automate TDex deamon with terraform
```
Comment (tiero) [inline blog/2021-07-13-tdex-terraform-automation.md:3]:
```suggestion
title: Deploy a TDEX Daemon to AWS with Terraform
```

File: blog/2021-07-13-tdex-terraform-automation.md [hunk 3]
```diff
@@ -0,0 +1,67 @@
+---
+slug: tdex-terraform-automation
```
Comment (tiero) [inline blog/2021-07-13-tdex-terraform-automation.md:2]:
```suggestion
slug: tdex-terraform-deploy
```

File: blog/2021-07-13-tdex-terraform-automation.md [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+title: Automate TDex deamon with terraform
+author: Alexander K
+author_title: Maintainer of TDEX Network
```
Comment (tiero) [inline blog/2021-07-13-tdex-terraform-automation.md:5]:
```suggestion
author_title: TDEX Contributor
```

File: blog/2021-07-13-tdex-terraform-automation.md [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+tags: [terraform, automation, tdex]
```
Comment (tiero) [inline blog/2021-07-13-tdex-terraform-automation.md:6]:
```suggestion
tags: [terraform, automation, tdex, deploy]
```

File: blog/2021-07-13-tdex-terraform-automation.md (hunk 1 above)
Comment (tiero) [inline blog/2021-07-13-tdex-terraform-automation.md:14]: Please would be nice to have screenshot of AWS console at least, ie. how to get secret/access key etc..

Better to describe each step with bullet points and separate sections for better readability

File: blog/2021-07-13-tdex-terraform-automation.md [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+  ssh_key_name: My Default Key \
+  IP Addr: Your IP \ 
+  Explorer URL: https://example.com/explorer \
+  S3 bucket name: my_backup_bucket
+
+Once you have everything in place, just execute deploy.sh with all the parameters in order provided above. 
+
+Example:
+```sh
+./deploy.sh aws_access_key aws_secret_key aws_region aws_ami ssh_public_key_path ssh_key_name IP_ADDR https://example.com/explorer aws_s3_my_backup_bucket
```
Comment (tiero) [inline blog/2021-07-13-tdex-terraform-automation.md:44]: Why not using read and making interactive prompt to ask each parameter? this is a bit clunky and prone to error for users with basci command line skills

File: blog/2021-07-13-tdex-terraform-automation.md [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+```
+
+
+Additionally, you can use unlockerd auto-unlock service, which is shipped with the container. 
+In the docker-compose file you can find commented lines for enabling the unlocker with the file provider, which means it attempts to source the unlocking password from a local file.
+
+Enabling the unlocker is as easy as creating a file containing the same password used to init your daemon's wallet and exporting its path in the PWD_PATH variable, like for example:
```
Comment (tiero) [inline blog/2021-07-13-tdex-terraform-automation.md:51]: Not clear where this should be exectuted. Will need the user to login via ssh in the remote machine etc..

File: blog/2021-07-13-tdex-terraform-automation.md [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+```sh
+$ echo "mypassword" > pwd.txt
+$ export PWD_PATH=$(pwd)/pwd.txt
+$ docker-compose up -d --no-deps --force-recreate tdexd
+```
+
+Execute unlockerd: 
+```sh
+$ docker exec -it tdexd unlockerd
```
Comment (tiero) [inline blog/2021-07-13-tdex-terraform-automation.md:62]: I would leave out the unlocker option, maybe is a bit difficult

**Review by tiero — CHANGES_REQUESTED** [review]
I think in general this blog post should be much more than some words on top of the tdex box, therefore we need to think about users with very entry level command line skills, so we dont need to leave out any steps and think they are known.

Also, we must be OPINIONATED and give out what we suggest as professional deployment, therefore we MUST have the S3 backup option enabled (and instructions on AWS if necessary) and most importantly the onion hidden service, so we can actually expose the tdex daemon to the world.

Last section should be about controlling the tdex daemons with the CLI (. ie. `tdex listmarket`) so an alias should be added

**Comment by kcoax** [comment]
I started working on changing the text to be more readable and will change the script to **read** inputs, not a big problem. 
I will also add support for Onion in automatic deployment. 

Regarding the controlling of tdex daemons etc, I think we might combine it with unlocked on a separate blog post? @tiero

**Comment by tiero** [comment]
> Regarding the controlling of tdex daemons etc, I think we might combine it with unlocked on a separate blog post? @tiero

Yes we can point to docs on how to init, unlock, deposit funds etc..

Poin is that user is running terraform from his local machine, is not quite obvious how from this blog post to go ahead and follow the docs on those matters, therefore we need to mention how the user controls. Simplest thing is to tell the user to SSH in the EC2 and use `tdex`alias (if we add automatically ie. `alias tdex="docker exec -it tdexd tdex "`)


Ohter option is to teach the user to install the `tdex` CLI in the local machine and do the macaroons pairing (more difficult maybe)

File: blog/2021-07-13-tdex-terraform-automation.md [hunk 9]
```diff
@@ -11,17 +11,21 @@ Configure and run TDex box easily using Terraform automation.
 <!--truncate-->
 
 
-As a first step it is required toobtain aws access and secret keys. Best practice is to create new user in AWS IAM. Proceed with opening AWS IAM users page, and click on Add user. Provide user with name, and below for the Access type select Programmatic access. Next, make sure that your AWS account(access key) has all required privileges to create EC2 instances and S3 access. 
+As a first step it is required toobtain aws access and secret keys. Best practice is to create new user in AWS IAM.
```
Comment (tiero) [inline blog/2021-07-13-tdex-terraform-automation.md]:
```suggestion
As a first step it is required to obtain aws access and secret keys. Best practice is to create new user in AWS IAM. 
```

File: blog/2021-07-13-tdex-terraform-automation.md [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+- - - - - 
+### Proceed with cloning and deploy
+Clone the TDex Box repository and enter its directory. 
+```sh
+git clone https://github.com/tdex-network/tdex-box.git
+cd tdex-box
+```
+- - - - - 
+
+On the AWS AMI Marketplace, find Ubuntu 18.04 public AMI, accessible in the region you are planing to deploy service and copy it's AMI ID.
```
Comment (tiero) [inline blog/2021-07-13-tdex-terraform-automation.md:46]: why not ubuntu 20?

File: blog/2021-07-13-tdex-terraform-automation.md [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+Clone the TDex Box repository and enter its directory. 
+```sh
+git clone https://github.com/tdex-network/tdex-box.git
+cd tdex-box
+```
+- - - - - 
+
+On the AWS AMI Marketplace, find Ubuntu 18.04 public AMI, accessible in the region you are planing to deploy service and copy it's AMI ID. 
+
+To run deploy please prepare: \
```
Comment (tiero) [inline blog/2021-07-13-tdex-terraform-automation.md:48]: It does not look good, maybe put in markdown quotes?

image

File: blog/2021-07-13-tdex-terraform-automation.md [hunk 12]
```diff
@@ -0,0 +1,69 @@
...
+  aws_ami: ami-05f7491af5eef733a \
+  ssh_public_key_path: ~/.ssh/id_rsa.pub \
+  ssh_key_name: My Default Key \
+  IP Addr: Your IP \ 
+  Explorer URL: https://example.com/explorer 
+
+Once you have everything in place, just execute deploy.sh and it will prompt you for all the parameters in order provided above. 
+Please enter parameters carefully. 
+```sh
+$ chmod +x deploy.sh
```
Comment (tiero) [inline blog/2021-07-13-tdex-terraform-automation.md:62]: is it in `terabox` folder? as per https://github.com/tdex-network/tdex-box/pull/16

**Comment by tiero** [comment]
Looks like you are pushing with wrong git user locally (ie. no PGP attached to that) @kcoax

**Comment by tiero** [comment]
still not looking good with markdown table

**Pull request merged**